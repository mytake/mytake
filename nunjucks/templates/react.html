{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
