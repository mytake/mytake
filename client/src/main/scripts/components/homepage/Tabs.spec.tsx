import * as React from "react";
import * as renderer from "react-test-renderer";
import Tabs from "./Tabs";

test("Tabs - expanded", () => {
  const props = {
    activeTab: {
      tabTitle: "How to use this",
    },
    onTabClick: jest.fn(),
    className: "testclass",
  };
  const tree = renderer.create(<Tabs {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
