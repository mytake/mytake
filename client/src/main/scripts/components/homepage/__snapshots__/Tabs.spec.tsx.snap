// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs - hidden 1`] = `
<div
  className="tabs"
>
  <div
    className="tab__column"
  >
    <ul
      className="tab__ul"
    >
      <li
        className="tab tab--active"
        onClick={[Function]}
      >
        How to use this
      </li>
      <li
        className="tab"
        onClick={[Function]}
      >
        What is this
      </li>
      <li
        className="tab"
        onClick={[Function]}
      >
        Get involved
      </li>
    </ul>
    <button
      className="tab__close"
      onClick={[MockFunction]}
    >
      <svg
        fill="none"
        height={24}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="15"
          x2="9"
          y1="9"
          y2="15"
        />
        <line
          x1="9"
          x2="15"
          y1="9"
          y2="15"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Tabs - visible 1`] = `
<div
  className="tabs tabs--visible"
>
  <div
    className="tab__column"
  >
    <ul
      className="tab__ul"
    >
      <li
        className="tab tab--active"
        onClick={[Function]}
      >
        How to use this
      </li>
      <li
        className="tab"
        onClick={[Function]}
      >
        What is this
      </li>
      <li
        className="tab"
        onClick={[Function]}
      >
        Get involved
      </li>
    </ul>
    <button
      className="tab__close"
      onClick={[MockFunction]}
    >
      <svg
        fill="none"
        height={24}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="15"
          x2="9"
          y1="9"
          y2="15"
        />
        <line
          x1="9"
          x2="15"
          y1="9"
          y2="15"
        />
      </svg>
    </button>
  </div>
</div>
`;
